OpenRA.Mods.Common/Activities/Move/MoveWithinRange.cs

```diff
@@ -47,8 +47,13 @@ protected override IEnumerable<CPos> CandidateMovementCells(Actor self)
 			var maxCells = (maxRange.Length + 1023) / 1024;
 			var minCells = minRange.Length / 1024;
 
+			// AtCorrectRange(map.CenterOfCell(c)) will return the current cell if the cell center is in range,
+			// even if the actor is actually out of range due to its current subcell not being the center.
+			// Avoid that by not including the current Location if the current actor.CenterPosition is out of range.
+			var ignoreCurrentCell = !AtCorrectRange(self.CenterPosition);
+
 			return map.FindTilesInAnnulus(targetPosition, minCells, maxCells)
-				.Where(c => AtCorrectRange(map.CenterOfCell(c)));
+				.Where(c => !(ignoreCurrentCell && c == self.Location) && AtCorrectRange(map.CenterOfCell(c)));
```

Comment:
A more correct solution here is to use `map.CenterOfSubCell(c, subCell)`, where you pass `subCell` in via `CandidateMovementCells(self, Mobile.FromSubCell);`